services/web/server/src/simcore_service_webserver/projects/projects_api.py

```diff
@@ -232,14 +253,27 @@ async def retrieve_and_notify_project_locked_state(
 
 
 @contextlib.asynccontextmanager
-async def lock_with_notification(
+async def lock_project_and_notify_state_update(
     app: web.Application,
     project_uuid: str,
     status: ProjectStatus,
     user_id: int,
     user_name: UserNameDict,
     notify_users: bool = True,
 ):
+    # FIXME: PC: I find this function very error prone. For instance, the requirements
+    # on the input parameters depend on the value of 'notify_users', i.e. changes dynamically.
+    #
+    # If notify_users=True, then project_uuid has to be defined in the database since
+    # the notification function the state which is in the database. On the other hand,
+    # locking relies on the project entry in redis.
+    # These two references to the project (redis and the db) are not in sync leading to some. An
+    # example is ``stop_servic``
+    # incosistent states that heavily depend on the logic of the function.
+    #
+    # Perhaps locking and notifications should be
```

Comment:
can you continue or remove this?

Reply:
finished sentence ... :-D and created #2872 to follow up as well